File: Tests/Oracle/corpus/sample.cpp
#include <string>
#include <vector>

namespace demo {

// A type the oracle should see as TypeLike.
class Widget {
  public:
    explicit Widget(std::string name) : name_(std::move(name)) {
    }

    const std::string& Name() const {
        return name_;
    }

  private:
    std::string name_;
};

int Total(const std::vector<int>& values) {
    int sum = 0;
    for (int v : values) {
        sum += v;
    }
    return sum;
}

} // namespace demo

File: Tests/Oracle/corpus/sample.ts
interface Sized {
    size(): number;
}

class Widget implements Sized {
    constructor(private name: string) {
    }

    size(): number {
        return this.name.length;
    }
}

const defaults = {
    width: 10,
    height: 20,
};

function total(values: number[]): number {
    let sum = 0;
    for (const v of values) {
        sum += v;
    }
    return sum;
}
